src/ws/test-server.c

```diff
@@ -632,7 +632,7 @@ on_handle_source (CockpitWebServer *server,
       inject_address (response, "bus_address", bus_address);
       inject_address (response, "direct_address", direct_address);
     }
-  cockpit_web_response_file (response, path, (const gchar **)server_roots);
+  cockpit_web_response_file_or_gz (response, TRUE, path, (const gchar **)server_roots);
```

Comment:
Could there be two unit test that make sure that serving gzip files actually works? (first one for accepting/receiving gz, the second for transparent decompression)

Reply:
There's actually three potential cases here:

 - request the .gz file explicitly, and want to receive .gz data
 - request the base file, with `Accept-Encoding: gzip` and expect to get the compressed stream
 - request the base file, with `Accept-Encoding: none` and make sure the transparent decompression happens

Do I understand correctly that you want to test the second two?

Reply:
Well, all cases should be covered/used somewhere -- otherwise why add the functionality in the first place? Or does this get implicitly covered by some later commit? (I didn't see any which would use that)

Reply:
FTR: as webpack now spits out .gz files, test-server does use them. So it's a preprequisite, and covered implicitly.